iothub_client/src/iothubtransport_mqtt_common.c

```diff
@@ -2254,6 +2254,15 @@ static int SendMqttConnectMsg(PMQTTTRANSPORT_HANDLE_DATA transport_data)
             }
         }
 
+        const char* dt_model_id = transport_data->transport_callbacks.dt_model_id_cb(transport_data->transport_ctx);
```

Comment:
>dt_model_id_cb [](start = 70, length = 14)

Please see my comment about potentially using the get_prodcut_info() as one-stop shop for this instead.

If we leave this explicit callback here, I believe that we would need to have this block of code in the isProductInfoSet block (or potentially in its own isDtSet, which could also argue for using the product_info() as only call).